pkg/operator/events/recorder_upstream.go

```diff
@@ -27,20 +31,63 @@ type upstreamRecorder struct {
 	broadcaster       record.EventBroadcaster
 	eventRecorder     record.EventRecorder
 	involvedObjectRef *corev1.ObjectReference
+	options           record.CorrelatorOptions
+
+	// shuttingDown indicates that the broadcaster for this recorder is being shut down
+	shuttingDown  bool
+	shutdownMutex sync.RWMutex
+
+	// fallbackRecorder is used when the kube recorder is shutting down
+	// in that case we create the events directly.
+	fallbackRecorder Recorder
+}
+
+var DefaultOperatorEventRecorderOptions = record.CorrelatorOptions{
```

Comment:
still forgot the keyFunc that needs to include the message to avoid de-duping different messages like we have today.

Reply:
keyFunc added